tests/testthat/test-step_naomit2.R

```diff
@@ -0,0 +1,30 @@
+library(dplyr)
+library(epiprocess)
+library(parsnip)
+library(workflows)
+
+# Random generated dataset
+set.seed(100)
+x <- tibble(geo_value = rep("nowhere",200),
+            time_value = as.Date("2021-01-01") + 0:199,
+            case_rate = rpois(100,20) + 1:200,
+            death_rate = rpois(100,10) + 1:200) %>%
+  as_epi_df()
+
+# Preparing the datasets to be used for comparison
+r <- epi_recipe(x) %>%
+  step_epi_ahead(death_rate, ahead = 7) %>%
+  step_epi_lag(death_rate, lag = c(0,7,14))
+
+z1 <- step_naomit2(r)
+z2 <- r %>%
+  step_naomit(all_predictors()) %>%
+  step_naomit(all_outcomes(), skip = TRUE)
+
+# Test
+test_that("Check that both functions behave the same way", {
+  expect_identical(z1$steps[[3]][-1][-5],z2$steps[[3]][-1][-5])
+  expect_identical(z1$steps[[4]][-1][-5],z2$steps[[4]][-1][-5])
+  expect_identical(class(z1$steps[[3]]),class(z2$steps[[3]]))
+  expect_identical(class(z1$steps[[4]]),class(z2$steps[[4]]))
```

Comment:
Again, why `[[4]]`?

See the comment above, but tests should ensure that other arguments don't break the functionality. What if someone calls `step_naomit2(..., someargs)`? The tests should cover anything that can possibly have the wrong downstream effects.

Reply:
@dajmcdon Could you elaborate please? I don't see any tests that test for using the wrong number of arguments. However, I did add a check so that you must use a recipe for this to work.

I could also add a functionality that makes it permissible to use a data frame, whereby the data frame would be converted to a recipe.

Reply:
Oh, I see. `step_naomit()` accepts many args whereas, as written, `step_naomit2()` accepts only a single argument, the recipe. I don't think it needs more functionality. And actually, we don't need these tests (because it's just a wrapper around already tested functions). 

However, it would be nice to make that much more explicit in the documentation. Also, to be consistent, the argument should be `recipe` rather than `x`.

Reply:
I kept the tests anyway. However, I updated the documentation to be more specific and checked if the input is a recipe.